docs/DeveloperGuide.md

```diff
@@ -1,13 +1,249 @@
+
+Demo: 
+
+`vis /t expense /c pie`
+
+Output
+
+`Displaying piechart for expense`
+A message will be shown telling you that the chart is being displayed
+
+![](images/vis/visOutput.png)
+
+This feature was implemented with the help of three different classes.
+They are namely: Visualizer, Categorizer, VisCommand (Inherits from abstract Command Class)
+
+VisCommand's Role: 
+1) Read the parameters of the vis command entered by the user
+- `/t` Reads the type of cashflow that the user wants to visualize (income/expense)
+- `/c` Reads the type of visualization tools the user wants (piechart/barchart)
+
+2) Calls the Cateorgizer to sort cashflow (Income/Expense) according to type
+
+3) Calls the Visualizer to display the chart to the user
+
+Categorizer's Role: 
+
+According to the cashflow type (Income/Expense) arugment passed in, the Categorizer sorts the 
+specified cashflow entry according to type using a Hashmap which is returned and used by the Visualizer
+
+Visualizer's Role: 
+
+According to the chart type (Pie/Bar) argument and the Hashmap obtained from the categorizer passed in, 
+the visualizer displays the specified visualization chart by calling the charting library Xchart.
+
+### Class Diagram
+
+![](images/vis/visualisationClass.png)
+
+### Sequence Diagram 
+
+Overall 
+
+![](images/vis/visualisationSequence.png)
+
+Categorizer
+
+![](images/vis/categorizerSequence.png)
+
+Visualizer
+
+![](images/vis/visualizerSequence.png)
+
+### Add income/expense feature
+
+The add income/expense command has 2 compulsory arguments `/t` and `/a` and 1 optional argument `/r`.
+
+Example:
+```
+add income /a 100 /t salary /r 30
+```
+Below are the steps that shows the implementation of add income/expense.
+#### Step 1
+An instantiated AddCashflowCommand class gets the instance of CashflowList.
+
+This allows the AddCashflowCommand instance to access the methods of CashflowList.
+#### Step 2
+The AddCashflowCommand instance then calls addIncome() or addExpense(), depending on what `category` is initialised as.
+
+addIncome() or addExpense() instantiates an Income or Expense object respectively.
+
+Example:
+```
+switch (category) {
+        case INCOME:
+            cashflowList.addIncome(amount, incomeType, recur);
+            break;
+        case EXPENSE:
+            cashflowList.addExpense(amount, expenseType, recur);
+            break;
+        default:
+            ui.showMessage("Unidentified entry.");
+            break;
+        }
+```
+#### Step 3
+The instantiated income/expense then updates the overall balance through addIncomeValue() or addExpenseValue().
+
+The income/expense object is also added to the list in Cashflowlist which contains all incomes/expenses.
+#### Step 4
+The added income/expense is then displayed to the user through the Ui.
+
+#### Diagrams
+Given below is the class diagram showing the class structure of the add income/expense mechanism:
+![](images/CashflowClassDiagram.png)
```

Comment:
Perhaps you will want to refer to the PlantUML docs to remove the C and E circles to comply with our standard of UML diagrams.